### PR TITLE
Sharing: Default to current site when fetching Publicize Connections

### PR DESCRIPTION
**Pull request by obenland**
This PR removes the implicit requirement to pass a site ID to retrieve Publicize connections.
While docs suggested that passing `siteId` is not required, it would send the request without it.

Not sure how bad of an idea it is to introduce the behavior of having a fallback in query components. Is there any potential fallout that you can think of?

In the readme I also removed the `query` prop passed to the component, I couldn't find any reference to it.

/cc @aduth as he did the only commit on this, @gwwar

### CODE REVIEW

**Comment by gwwar** [comment]
> Not sure how bad of an idea it is to introduce the behavior of having a fallback in query components. Is there any potential fallout that you can think of?

The only case I can think of is if you're attempting to query site B but your currently connected site is site A. The Site B object might not be available, but Site A is, so the initial request will return Site A connections. I don't think we'd ever have this use case though.

Looking at it I also wouldn't expect it to fallback to the current site without looking at docs.

The code looks fine here, but are there any use cases that makes this easier to use by allowing an optional siteId? Maybe we could specify a boolean attr if we want the current selected site. e.g
```javascript
<QueryPublicizeConnections selectedSite />
```

**Comment by obenland** [comment]
> The Site B object might not be available, but Site A is, so the initial request will return Site A connections.

The component doesn't need the entire site object, only the site id. This is also the way it currently works

> Looking at it I also wouldn't expect it to fallback to the current site without looking at docs.

What would you expect it to fall back to?

**Comment by gwwar** [comment]
It's a bit contrived but I'm talking about the falsey case where we wait for things to load. The intent might be mixed up. We ran into this for #7844 but that was a much more likely to happen.

```jsx
// in render
<QueryPublicizeConnections siteId={ siteId } /> // now attempts to fetch current site id
//  in connect:
const siteId = siteB ? siteB.id || null;
```

I'm just suggesting that we leave the interface such that we don't have to think about falsey cases.
```
<QueryPublicizeConnections siteId={ siteId } /> //only fetches that site
<QueryPublicizeConnections selectedSite /> //will attempt to fetch the current site
```

**Comment by obenland** [comment]
Ah, I understand. Yes, that makes sense

**Comment by obenland** [comment]
Fixed in dcdeaefa1d2f1325e4dd275e00c30fca4997ab98

File: client/components/data/query-publicize-connections/README.md [hunk 1]
```diff
@@ -5,7 +5,7 @@ Query Publicize Connections
 
 ## Usage
 
-Render the component, passing `siteId`. It does not accept any children, nor does it render any elements to the page. You can use it adjacent to other sibling components which make use of the fetched data made available through the global application state.
+Render the component, passing `siteId` or `selectedSite`. It does not accept any children, nor does it render any elements to the page. You can use it adjacent to other sibling components which make use of the fetched data made available through the global application state.
```
Comment (gwwar) [inline client/components/data/query-publicize-connections/README.md]: Maybe note that selectedSite is a boolean attr.

File: client/components/data/query-publicize-connections/index.jsx [hunk 2]
```diff
@@ -34,24 +34,29 @@ class QueryPublicizeConnections extends Component {
...
+	requestConnections: () => {},
+	requestingConnections: false,
+	selectedSite: false,
+	siteId: 0,
 };
 
 export default connect(
-	( state, ownProps ) => {
+	( state, { siteId, selectedSite } ) => {
+		siteId = siteId || ( selectedSite && getSelectedSiteId( state ) );
```
Comment (gwwar) [inline client/components/data/query-publicize-connections/index.jsx]: How about the inverse? If someone provides both args we'll always just fetch the selectedSite.
```jsx
const selectedSiteId = selectedSite && getSelectedSiteId( state );
const siteId = selectedSiteId || siteId;
```

File: client/components/data/query-publicize-connections/index.jsx [hunk 3]
```diff
@@ -34,24 +34,29 @@ class QueryPublicizeConnections extends Component {
...
 QueryPublicizeConnections.propTypes = {
-	siteId: PropTypes.number,
+	requestConnections: PropTypes.func,
 	requestingConnections: PropTypes.bool,
-	requestConnections: PropTypes.func
+	selectedSite: PropTypes.bool,
+	siteId: PropTypes.number,
 };
 
 QueryPublicizeConnections.defaultProps = {
```
Comment (gwwar) [inline client/components/data/query-publicize-connections/index.jsx]: Let's take this opportunity to bulletproof our lifecycle methods. We should update to `componentDidMount` and `componentDidUpdate`

See also: https://gist.github.com/bvaughn/923dffb2cd9504ee440791fade8db5f9

File: client/components/data/query-publicize-connections/index.jsx (hunk 2 above)
Comment (obenland) [inline client/components/data/query-publicize-connections/index.jsx]: I think if there's a siteId specified it should take precedent over a selected site. It feels more intentional

File: client/components/data/query-publicize-connections/index.jsx [hunk 4]
```diff
@@ -3,29 +3,25 @@
...
 		}
 	}
 
-	componentWillReceiveProps( nextProps ) {
-		if ( nextProps.requestingConnections ||
-				! nextProps.siteId ||
-				( this.props.siteId === nextProps.siteId ) ) {
-			return;
+	componentDidUpdate( { siteId } ) {
+		if ( siteId !== this.props.siteId && ! this.props.requestingConnections ) {
```
Comment (gwwar) [inline client/components/data/query-publicize-connections/index.jsx]: Let's also check to see if siteId is falsey

Comment (gwwar) [inline client/components/data/query-publicize-connections/index.jsx]: Feel free to 🚢 after ^^ has been updated

**Pull request merged**